src/cpp/dataset.cpp

```diff
@@ -367,3 +370,41 @@ TensorBase* DataSet::_get_tensorbase_obj(const std::string& name)
     _enforce_tensor_exists(name);
     return _tensorpack.get_tensor(name)->clone();
 }
+
+// Create a string representation of the DataSet
+std::string DataSet::to_string() const
+{
+    std::string result;
+    result = "DataSet (" + _lname + "):\n";
+
+    // Tensors
+    result += "Tensors:\n";
```

Comment:
Very minor, but if number or tensors or number of metadata fields is 0, should we print something like ``None`` under the heading:

```
Metadata:
  None
```

So the message looks more complete when one is empty?

Reply:
That's a great suggestion!